### PR TITLE
OBSDATA-1697: Do not build extensions not loaded by cc-druid

### PR DESCRIPTION
**Pull request by janjwerner-confluent**
### Description
Create new profiles to enable only the used extensions during the build. This helps address CVEs that were being flagged due to the unused extensions.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -158,64 +158,66 @@
...
+        <!-- <module>extensions-core/stats</module> -->
+        <!-- <module>extensions-core/kafka-extraction-namespace</module> -->
         <module>extensions-core/kafka-indexing-service</module>
-        <module>extensions-core/kinesis-indexing-service</module>
-        <module>extensions-core/multi-stage-query</module>
-        <module>extensions-core/mysql-metadata-storage</module>
-        <module>extensions-core/orc-extensions</module>
-        <module>extensions-core/parquet-extensions</module>
+        <!-- <module>extensions-core/kinesis-indexing-service</module> -->
+        <!-- <module>extensions-core/multi-stage-query</module> -->
```
Comment (xvrl) [inline pom.xml:174]: we should leave multi-stage-query enabled, it's a core extension which doesn't pull in any additional dependencies and brings lots of useful functionality

File: pom.xml [hunk 2]
```diff
@@ -158,64 +158,66 @@
...
+        <!-- <module>extensions-core/mysql-metadata-storage</module> -->
+        <!-- <module>extensions-core/orc-extensions</module> -->
+        <!-- <module>extensions-core/parquet-extensions</module> -->
         <module>extensions-core/postgresql-metadata-storage</module>
-        <module>extensions-core/protobuf-extensions</module>
-        <module>extensions-core/lookups-cached-global</module>
-        <module>extensions-core/lookups-cached-single</module>
-        <module>extensions-core/ec2-extensions</module>
+        <!-- <module>extensions-core/protobuf-extensions</module> -->
```
Comment (xvrl) [inline pom.xml:179]: the protobuf extensions are useful to have to debug data streams, we already depend on the same dependencies in our other modules.

File: pom.xml [hunk 3]
```diff
@@ -158,64 +158,66 @@
...
+        <!-- <module>extensions-core/lookups-cached-global</module> -->
+        <!-- <module>extensions-core/lookups-cached-single</module> -->
+        <!-- <module>extensions-core/ec2-extensions</module> -->
         <module>extensions-core/s3-extensions</module>
         <module>extensions-core/druid-aws-rds-extensions</module>
-        <module>extensions-core/simple-client-sslcontext</module>
+        <!-- <module>extensions-core/simple-client-sslcontext</module> -->
```
Comment (xvrl) [inline pom.xml:185]: we may need this for fedramp, so let's leave it in

File: pom.xml [hunk 4]
```diff
@@ -158,64 +158,66 @@
         <module>cloud/aws-common</module>
         <module>cloud/gcp-common</module>
         <!-- Core extensions -->
-        <module>extensions-core/kubernetes-extensions</module>
-        <module>extensions-core/avro-extensions</module>
-        <module>extensions-core/azure-extensions</module>
+        <!-- <module>extensions-core/kubernetes-extensions</module> -->
```
Comment (xvrl) [inline pom.xml:161]: we will need this soon

File: pom.xml [hunk 5]
```diff
@@ -158,64 +158,66 @@
...
-        <module>extensions-contrib/aliyun-oss-extensions</module>
-        <module>extensions-contrib/prometheus-emitter</module>
-        <module>extensions-contrib/opentelemetry-emitter</module>
-        <module>extensions-contrib/kubernetes-overlord-extensions</module>
+        <!-- <module>extensions-contrib/time-min-max</module> -->
+        <!-- <module>extensions-contrib/virtual-columns</module> -->
+        <!-- <module>extensions-contrib/thrift-extensions</module> -->
+        <!-- <module>extensions-contrib/ambari-metrics-emitter</module> -->
+        <!-- <module>extensions-contrib/sqlserver-metadata-storage</module> -->
+        <!-- <module>extensions-contrib/kafka-emitter</module> -->
```
Comment (xvrl) [inline pom.xml:207]: we are using kafka-emitter

File: pom.xml [hunk 6]
```diff
@@ -158,64 +158,66 @@
...
+        <!-- <module>extensions-contrib/materialized-view-maintenance</module> -->
+        <!-- <module>extensions-contrib/materialized-view-selection</module> -->
+        <!-- <module>extensions-contrib/momentsketch</module> -->
+        <!-- <module>extensions-contrib/moving-average-query</module> -->
+        <!-- <module>extensions-contrib/tdigestsketch</module> -->
+        <!-- <module>extensions-contrib/influxdb-emitter</module> -->
+        <!-- <module>extensions-contrib/gce-extensions</module> -->
+        <!-- <module>extensions-contrib/aliyun-oss-extensions</module> -->
+        <!-- <module>extensions-contrib/prometheus-emitter</module> -->
+        <!-- <module>extensions-contrib/opentelemetry-emitter</module> -->
```
Comment (xvrl) [inline pom.xml:219]: we are using opentelemetry-emitter

File: pom.xml [hunk 7]
```diff
@@ -158,64 +158,66 @@
...
+        <!-- <module>extensions-contrib/kubernetes-overlord-extensions</module> -->
```
Comment (xvrl) [inline pom.xml:220]: we will need this soon

**Comment by janjwerner-confluent** [comment]
@xvrl 
Thank you for all the feedback - I used the description of loaded modules and cut as deep as I could. I will apply the comments you placed and try again.

**Comment by janjwerner-confluent** [comment]
@xvrl 
re-enabled the requested extensions. clean, succesful build.

**Comment by xvrl** [comment]
@janjwerner-confluent rather than disabling compilation of modules I would prefer if we omitted those modules from the packaging step in https://github.com/confluentinc/druid/blob/25.0.0-confluent/distribution/pom.xml#L584-L641. Even better, we should create a separate profile that enables only the subset of extensions we need, and use that.

This will make it easier to customize what we build without impacting our ability to run integration tests, and ensure we can still test locally with other extensions using the default dist,bundle-contrib profiles.

**Comment by KeerthanaSrikanth** [comment]
> @janjwerner-confluent rather than disabling compilation of modules I would prefer if we omitted those modules from the packaging step in https://github.com/confluentinc/druid/blob/25.0.0-confluent/distribution/pom.xml#L584-L641. Even better, we should create a separate profile that enables only the subset of extensions we need, and use that.
> 
> This will make it easier to customize what we build without impacting our ability to run integration tests, and ensure we can still test locally with other extensions using the default dist,bundle-contrib profiles.

@xvrl 
I have tried to address this comment and added new profiles. Is this what you meant? Or did you have something else in mind?

**Review by m-ghazanfar — APPROVED** [review]

**Pull request merged**